src/AzureAuth/CommandInfo.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.AzureAuth
+{
+    using System.IO.Abstractions;
+    using System.Reflection;
+    using McMaster.Extensions.CommandLineUtils;
+    using Microsoft.Extensions.Logging;
+    using Microsoft.Office.Lasso.Telemetry;
+
+    /// <summary>
+    /// The command <see cref="CommandInfo"/> shows debug information and system runtime.
+    /// </summary>
+    [Command(Name = "info", Description = "Shows AzureAuth debug information. Please provide when asking for help.")]
+    public class CommandInfo
+    {
+        private const string OptionResetDeviceID = "--reset-device-id";
+        private readonly ILogger<CommandInfo> logger;
+        private readonly IFileSystem fileSystem;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CommandInfo"/> class.
+        /// </summary>
+        /// <param name="logger">The logger.</param>
+        /// <param name="fileSystem">The file system.</param>
+        public CommandInfo(ILogger<CommandInfo> logger, IFileSystem fileSystem)
+        {
+            this.logger = logger;
+            this.fileSystem = fileSystem;
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether reset device id.
+        /// </summary>
+        [Option(OptionResetDeviceID, "Reset Device ID", CommandOptionType.NoValue)]
+        public bool ResetDeviceID { get; set; }
+
+        /// <summary>
+        /// This method executes the info process.
+        /// </summary>
+        /// <returns>The error code: 0 is normal execution, and the rest means errors during execution.</returns>
+        public int OnExecute()
+        {
+            if (this.ResetDeviceID)
+            {
+                return this.ExecuteResetDeviceID();

```

Comment:
Last major design question about this - we are doing a different action given the presence of a flag. This is a design smell that I know I'm guilty of making in the past. 

If the user knows they want to function of the flag, and the flag is not modifying the actual command being run. Then you likely don't want to use a flag, what you have is another command. Perhaps a sub-command in this case. 

This is the same kind of code smell as passing a bool into a function, and then doing thing A if true, otherwise thing B if false. In that case - you can simplify by writing functionA and functionB, which take no parameters.

So, I dislike using a flag to change the entire logic of the command. I think it will be more simple and clear if we either:
1. Make the flag _modify_ the current command. meaning if set, do reset the device if, and then proceed with the rest of the command printing the info, or
2. Make the reset device id a separate, sub-command on the info command so that we have
   ```shell
   azureauth info -> prints info
   azureauth info reset-device-id -> only resets device id.
   ```

Reply:
@reillysiemens @goagain what do you think about sub-command vs modifier for resetting device id?

Reply:
I really thought about this question. Both options work for me. The answer why I took `--reset-device-id` was `--clear` did the same thing in CommandMain.

Reply:
That makes sense, and unfortunately in this case, `--clear` is also an example of this design smell which I regret. The reason `--clear` is attached to the command is that in order to clear, at present, we still need to make a Public client application instance with valid parameters, which we get on that command. I will open an issue to track simplifying the clear function.

Reply:
If you are fine with either option. I think having this flag as a modifer, so clear it, and then immeditaely regnerate and print the new one works, because ten someone can easily see that it's changing without needing more UI from another command.

Reply:
I am glad to see things go to consistency.